### PR TITLE
feature(typescript_indexer): add Kythe typings

### PR DESCRIPTION
**Pull request by ayazhafiz**
Add typings for Kythe. Ported from
[@types/kythe](https://www.npmjs.com/package/@types/kythe).

### CODE REVIEW

**Comment by ayazhafiz** [comment]
cc @evmar 

I put a function to create ordinal edges in the typings file, so it's not true `.d.ts`. Let me know if that should moved elsewhere.

File: kythe/typescript/kythe.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
+/**
```
Comment (evmar) [inline kythe/typescript/kythe.ts]: Add a @fileoverview at the top describing what this is.  Which I think is something like "exposes to TS the types and values from the kythe schema", or some such.

File: kythe/typescript/kythe.ts [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+ * A VName (Vector Name) for a node in the Kythe schema consists of:
+ *  - `signature`: a unique, opaque signature for a node
+ *  - `corpus`: a collection of related files the node is defined in
+ *  - `root`: a label denoting a distinct subset of the corpus
+ *  - `path`: the relative path of the file containing the node
+ *  - `language`: programming language the node belongs to
+ */
+export interface VName {
```
Comment (evmar) [inline kythe/typescript/kythe.ts]: Shouldn't this be used by indexer.ts?

File: kythe/typescript/kythe.ts [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+ *   `${EdgeKind}.${number}`
+ * This is represented as a branded string that is incompatible with a string
+ * but can be compared to a string.
+ */
+export type OrdinalEdge = string&{
+  __ordinalBrand: 'ordinal';
+};
+export function makeOrdinalEdge(edge: EdgeKind, ordinal: number): OrdinalEdge {
+  const ordinalEdge = `${edge}.${ordinal}`;
+  return <OrdinalEdge>ordinalEdge;
```
Comment (evmar) [inline kythe/typescript/kythe.ts]: I think
return ordinalEdge as OrdinalEdge;
The brackets syntax is obsolete

File: kythe/typescript/kythe.ts [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+ * A Fact is an Entry that also has a fact `value`.
+ */
+export interface Fact extends Entry {
+  value: string;
+}
+
+/**
+ * An Edge is an Entry that also has a `target` and an edge `kind`.
+ */
+export interface Edge extends Entry {
```
Comment (evmar) [inline kythe/typescript/kythe.ts]: Should these be used by indexer.ts too?

Comment (ayazhafiz) [reply kythe/typescript/kythe.ts]: Yes, I'm going to add that in a separate PR.

File: kythe/typescript/kythe.ts (hunk 2 above)
Comment (ayazhafiz) [reply kythe/typescript/kythe.ts]: Yes, going to add that in a separate PR.

File: kythe/typescript/kythe.ts (hunk 1 above)
Comment (ayazhafiz) [reply kythe/typescript/kythe.ts]: Done.

File: kythe/typescript/kythe.ts (hunk 3 above)
Comment (ayazhafiz) [reply kythe/typescript/kythe.ts]: Done.

File: kythe/typescript/kythe.ts [hunk 5]
```diff
@@ -0,0 +1,211 @@
+/**
+ * @fileoverview TypeScript typings of the Kythe schema.
```
Comment (evmar) [inline kythe/typescript/kythe.ts]: 'typings' has a specific meaning in TS, it's about defining types for a JS library, which I think doesn't apply here.

File: kythe/typescript/kythe.ts [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+ * @author Kythe Authors
```
Comment (evmar) [inline kythe/typescript/kythe.ts]: I think this is can be omitted.

Now that I think of it, I think you need to copy the license block into this file too.
So like

/*
 * Copyright [paste from other file ]
 */

/** @fileoverview TypeScript definitions of types found in the Kythe schema. */

Comment (ayazhafiz) [reply kythe/typescript/kythe.ts]: Done.

File: kythe/typescript/kythe.ts (hunk 5 above)
Comment (ayazhafiz) [reply kythe/typescript/kythe.ts]: Fixed.

**Comment by ayazhafiz** [comment]
@kythe PTAL

**Pull request merged**